src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotPrimaryDataQueryUtils.java

```diff
@@ -0,0 +1,46 @@
+package bio.terra.service.snapshot.flight.create;
+
+import bio.terra.grammar.Query;
+import bio.terra.grammar.exception.InvalidQueryException;
+import bio.terra.service.dataset.AssetSpecification;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.snapshot.exception.AssetNotFoundException;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class CreateSnapshotPrimaryDataQueryUtils {
```

Comment:
Note for code reviewers: I struggled a bit with where to put the code that is shared between the GCP and Azure versions of snapshot by query. As far as I could tell, there wasn't a good way to use an inherited class because we're already inheriting CreateSnapshotParquetFilesAzureStep for the azure step. So, I settled with a utils method.

Reply:
I tried going down the interface route, but I ran into issues there, too (Unfortunately, I can't remember exactly what). 

But, looking at the broader picture, for interfaces, you're really just defining the methods you want to be implemented. For these utils methods, I want to share a few methods between the GCP & Azure steps so that I don't have to re-implement them. It's really better suited for `extends class`.  Do you agree? Is there some way to do an interface that works in this situation?

Reply:
I think you could do something with a `default` method?  We can talk at standup

Reply:
Ah, okay, I see now! I'm working on this approach.